tcms/testruns/tests/test_views.py

```diff
@@ -579,6 +590,21 @@ def test_edit_run(self):
 
         self.assertRedirects(response, reverse('testruns-get', args=[run.pk]))
 
+    def test_user_without_permissions_should_not_be_able_to_edit(self):
+        remove_perm_from_user(self.tester, 'testruns.change_testrun')
+
+        post_data = {
+            'summary': 'New run summary',
+            'build': self.new_build.pk,
+            'manager': self.test_run.manager.email,
+            'default_tester': self.intern.email,
+            'notes': 'easytest',
+        }
+
+        self.assertRedirects(
+            self.client.post(self.edit_url, post_data),
+            reverse('tcms-login') + '?next=' + self.edit_url)
```

Comment:
See `bugs/tests/test_permissions.py` - add this as a separate TC with a similar syntax. Do this before continuing with the refactoring or afterwards, up to you.